src/Service/RegisteredMethodManager.php

```diff
@@ -104,6 +105,33 @@ public function registerForMember(Member $member, MethodInterface $method, $data
         return true;
     }
 
+    /**
+     * Determines if a method can be removed
+     *
+     * By default this is false if MFA is required and the method is the last on the Member (besides the backup method)
+     * but the funcation provides a hook point for extensibility e.g. if an site requires a particular method to be in
+     * use by a subset of members - admins must use U2F but normal users can use TOTP.
+     *
+     * @param Member $member
+     * @param MethodInterface $method
+     * @return bool
+     */
+    public function canRemoveMethod(Member $member, MethodInterface $method): bool
+    {
+        $removable = true;
+        $remainingMethods = $member->RegisteredMFAMethods()->count();
+        $mfaIsRequired = Injector::inst()->get(EnforcementManager::class)->isMFARequired();
+
+        // This is the last method (besides the backup method), and MFA is required
+        if ($mfaIsRequired && $remainingMethods === 2) {
```

Comment:
There is a possibility (with configuration) that you don't have a backup method - this means this check is wrong. You need to assert that they have a backup method registered before checking `2` - otherwise it'll be just `1`.